uber/site_sections/registration.py

```diff
@@ -500,7 +500,7 @@ def register(self, session, message='', **params):
                 message = 'Please select a payment type'
             elif attendee.payment_method == c.MANUAL and not re.match(c.EMAIL_RE, attendee.email):
                 message = 'Email address is required to pay with a credit card at our registration desk'
-            elif attendee.badge_type not in [c.ATTENDEE_BADGE, c.ONE_DAY_BADGE]:
+            elif attendee.badge_type not in [badge for badge, desc in c.AT_THE_DOOR_BADGE_OPTS]:
```

Comment:
Shouldn't be too hard to reassociate them.  We currently have ``c.BADGE_OPTS`` and ``c.BADGE_VARS`` properties which looks like this

```python
>>> c.BADGE_OPTS
[(51352218, 'Attendee'), (67489953, 'Supporter'), (16863825, 'Staff'), (144001217, 'Guest'), (71138939, 'One Day')]
```

So if we add these new badge types

```
[enums]
[[badge]]
thursday = "Thursday"
friday = "Friday"
saturday = "Saturday"
```

Then it'll look like

```python
>>> c.BADGE_OPTS
[(51352218, 'Attendee'), (67489953, 'Supporter'), (16863825, 'Staff'), (144001217, 'Guest'), (71138939, 'One Day'), (176673583, 'Thursday'), (84680830, 'Friday'), (62603428, 'Saturday')]
```

So we could rewrite ``can_check_in`` 

```python
@property
def can_check_in(self):
    return self.paid != c.NOT_PAID and self.badge_status in [c.NEW_STATUS, c.COMPLETED_STATUS] and not self.is_unassigned
```

to something like this

```python
@property
def can_check_in(self):
    valid = self.paid != c.NOT_PAID and self.badge_status in [c.NEW_STATUS, c.COMPLETED_STATUS] and not self.is_unassigned
    if valid and self.badge_type_label in {'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'}:
        valid = self.badge_type_label == localized_now().strftime('%A')
    return valid
```

The only downside is that it requires that we MUST have the badge type labels match the day of week name.  I think that's fine, but it's something we'd want to document in ``configspec.ini``.